tools/loader/Loader.cpp

```diff
@@ -401,11 +447,30 @@ Loader::Loader(int argc, char **argv) {
     caffe2NetWeightFilename_ = modelPathOpt[1];
   }
   M_.reset(new Module);
+
   std::vector<std::unique_ptr<runtime::DeviceConfig>> configs;
-  for (unsigned int i = 0; i < numDevices; ++i) {
-    auto config = llvm::make_unique<runtime::DeviceConfig>(ExecutionBackend);
-    configs.push_back(std::move(config));
+
```

Comment:
This adds a lot of logic to a function that is already long. Is there a way to refactor things and make the code more readable?

Reply:
Get this piece of code into a function.